src/Bash.cpp

```diff
@@ -0,0 +1,297 @@
+        _currPipe[0] = other._currPipe[0];
+        _currPipe[1] = other._currPipe[1];
+
+        // Copy the input/output file descriptors
+        _prevPipe[0] = other._prevPipe[0];
+        _prevPipe[1] = other._prevPipe[1];
+    }
+    return *this;
+}
+
+/**
+ * Move assignment operator for the Bash class.
+ * Moves the queue of ICommand pointers, the vector of path directories,
+ * and the pipe file descriptors from the source object to the destination object.
+ * @param other The Bash object to be moved.
+ * @return The reference to the destination object.
+ */
+Bash& Bash::operator=(Bash&& other) noexcept {
+    if (this == &other) {
+        return *this;  // Self-assignment check
+    }
+
+    // Free the resources held by the destination object
+    while (!_commands.empty()) {
+        delete _commands.front();
+        _commands.pop();
+    }
+
+    // Move the resources from the source object to the destination object
+    _commands = std::move(other._commands);
+    _pathDirectories = std::move(other._pathDirectories);
+    _currPipe[0] = other._currPipe[0], _currPipe[1] = other._currPipe[1];
+    _prevPipe[0] = other._prevPipe[0], _prevPipe[1] = other._prevPipe[1];
+
+    // Set the source object's members to a valid state or nullify them
+    other._currPipe[0] = 0;
+    other._currPipe[1] = 0;
+    other._prevPipe[0] = 0;
+    other._prevPipe[1] = 0;
+
+    return *this;
+}
+
+/**
+ * Start the execution of commands based on the given input.
+ * Parses the input into individual commands and sets up the necessary pipes.
+ * Executes each command sequentially and waits for the child process to finish before executing the next command.
+ * @param input The input string containing the commands.
+ * @return 0 on success.
+ */
+int Bash::start(const string& input) {
+    _prevPipe[0] = -1;
+    _prevPipe[1] = -1;
+
+    // Split the input into commands and search for pipe sections
+    vector<string> commandStrings = splitString(input, " | ");
+    size_t redirectionIndex;
+    // Create the Command instances
+    for (const string& commandStr : commandStrings) {
+
+        vector<string> args = splitString(commandStr, " ");
+
+        if ((redirectionIndex = findRedirection(args, ">")) != 0)
+            _commands.emplace(new RedirectionCommand(args, _pathDirectories, redirectionIndex));
+
+        else if ((redirectionIndex = findRedirection(args, "<")) != 0) {
+            _commands.emplace(new RedirectionCommand(args, _pathDirectories, redirectionIndex));
+        }
+        else {
+            _commands.emplace(new Command(args, _pathDirectories));
+        }
+    }
+
+    // Execute each command sequentially
+    while (!_commands.empty()) {
+        ICommand* currCommand = _commands.front();
+        _commands.pop();
+
+        // change directory
+        if (currCommand->getCommand() && strcmp(currCommand->getCommand()[0], "cd") == 0) {
+            if (chdir(*(currCommand->getCommand() + 1)) == -1) {
```

Comment:
`chdir` is old fashioned C style way. Use the modern C++ way with `std::filesystem::current_path`